third_party/move/move-vm/runtime/src/storage/environment.rs

```diff
@@ -0,0 +1,175 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    config::VMConfig,
+    native_functions::{NativeFunction, NativeFunctions},
+    storage::{struct_name_index_map::StructNameIndexMap, verifier::VerifierExtension},
+    Module, Script,
+};
+use move_binary_format::{
+    access::ModuleAccess, errors::PartialVMResult, file_format::CompiledScript, CompiledModule,
+};
+use move_bytecode_verifier::dependencies;
+use move_core_types::{
+    account_address::AccountAddress,
+    identifier::{IdentStr, Identifier},
+};
+use std::sync::Arc;
+
+/// Wrapper around partially verified compiled module, i.e., one that passed
+/// local bytecode verification, but not the dependency checks yet. Also
+/// carries size in bytes.
+pub struct PartiallyVerifiedModule(Arc<CompiledModule>, usize);
```

Comment:
another possible name is LocallyVerifiedModule - not sure if better, partial is also informative and good